engine/flink/avro-util/src/main/scala/pl/touk/nussknacker/engine/avro/KafkaAvroSourceFactory.scala

```diff
@@ -53,9 +54,14 @@ class KafkaTypedAvroSourceFactory[T: TypeInformation](schemaFactory: Deserializa
              @ParamName("schema")
              @NotBlank
              avroSchema: String): Source[T] with TestDataGenerator = {
-    val schemaRegistryClient = schemaRegistryClientFactory.createSchemaRegistryClient(kafkaConfig)
```

Comment:
Be aware that schema registry client is a fat object. You should rather think about like a database connection pool. The best would be have it created just once for whole task manager. Because of flink's limitation of services (availabled in NiFi) we have some duplication of resources. But we should keep this duplication as lower as possible. It looks like you've created this client many times in `schemaRegistryProvider`

Reply:
I understud that. But now after these changes client is created only once time at `ConfluentSchemaRegistryProvider.def apply[T: TypeInformation](processObjectDependencies: ProcessObjectDependencies, useSpecificAvroReader: Boolean)`I understud that. But now after these changes client is creted only once time at `ConfluentSchemaRegistryProvider.def apply[T: TypeInformation](processObjectDependencies: ProcessObjectDependencies, useSpecificAvroReader: Boolean)`

Reply:
Now I user SchemaRegistryClientFactory.